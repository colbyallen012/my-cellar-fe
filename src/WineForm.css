.wine-input{
  background-color: white;
  width: 50%;
}

input {
  width: 98%;
}