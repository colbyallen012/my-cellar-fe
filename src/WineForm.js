import React, { Component } from 'react';
import { postWine } from './ApiCalls';
import './WineForm.css';

export class WineForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      vineyard: '',
      name: '',
      color: '',
      type: '',
      year: '',
      rating: '',
    }
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({ [name]: value})
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    await postWine(this.state)
    this.setState({vineyard: '', name: '', color: '', type: '', year: '', rating: ''})
  }

  render() {
    return(
      <div className='WineForm'>
        <form className='wine-input' onSubmit={this.handleSubmit}>
          <input
            type='text'
            value={this.state.vineyard}
            name='vineyard'
            placeholder='Vineyard'
            onChange={this.handleChange}
          />
          <input
            type='text'
            value={this.state.name}
            name='name'
            placeholder='Name'
            onChange={this.handleChange}
          />
          <input
            type='text'
            value={this.state.color}
            name='color'
            placeholder='Color'
            onChange={this.handleChange}
          />
          <input
            type='text'
            value={this.state.type}
            name='type'
            placeholder='Type'
            onChange={this.handleChange}
          />
          <input
            type='number'
            value={this.state.year}
            name='year'
            placeholder='Year'
            onChange={this.handleChange}
          />
          <input
            type='rating'
            value={this.state.rating}
            name='rating'
            placeholder='Rating'
            onChange={this.handleChange}
          />
          <button>Add Wine</button>
        </form>
      </div>
    )
  }
}

export default WineForm