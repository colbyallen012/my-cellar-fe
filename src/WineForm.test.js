import React from 'react';
import { shallow } from 'enzyme'
import WineForm from './WineForm'

describe('WineForm', () => {
  let wrapper, instance;
  const props = {
    getWines: jest.fn()
  }

  beforeEach(() => {
    wrapper = shallow(<WineForm {...props}/>)
    instance = wrapper.instance()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})