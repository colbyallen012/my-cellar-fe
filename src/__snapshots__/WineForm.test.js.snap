// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WineForm should match snapshot 1`] = `
<div
  className="WineForm"
>
  <h2
    className="add-wine"
  >
    Add Wine
  </h2>
  <form
    className="wine-input"
    onSubmit={[Function]}
  >
    <input
      name="vineyard"
      onChange={[Function]}
      placeholder="Vineyard"
      type="text"
      value=""
    />
    <input
      name="name"
      onChange={[Function]}
      placeholder="Bottle Name"
      type="text"
      value=""
    />
    <input
      name="color"
      onChange={[Function]}
      placeholder="Color"
      type="text"
      value=""
    />
    <input
      name="type"
      onChange={[Function]}
      placeholder="Type"
      type="text"
      value=""
    />
    <input
      name="year"
      onChange={[Function]}
      placeholder="Year"
      type="number"
      value=""
    />
    <input
      name="rating"
      onChange={[Function]}
      placeholder="Rating"
      type="number"
      value=""
    />
    <button
      className="add"
    >
      Submit
    </button>
  </form>
</div>
`;
